py-polars/polars/internals/frame.py

```diff
@@ -2223,7 +2223,21 @@ def schema(self) -> dict[str, type[DataType]]:
         {'foo': <class 'polars.datatypes.Int64'>, 'bar': <class 'polars.datatypes.Float64'>, 'ham': <class 'polars.datatypes.Utf8'>}
 
         """
-        return dict(zip(self.columns, self.dtypes))
+        # to recursively pull nested dtypes
```

Comment:
I think we should properly create the nested dtype on the rust side.

Reply:
Sounds fun :) 

Would we just update the display implementation of `DataType`?
https://github.com/pola-rs/polars/blob/9344555b3133626b56a5bf6e207d347f9badbd8e/polars/polars-core/src/datatypes/mod.rs#L540

Reply:
Since I wasn't planning on this being super involved I'll check his out passively.

I'm no longer thinking it's just Display that needs to be update, but maybe the series.dtype function itself is enough.

https://github.com/pola-rs/polars/blob/9344555b3133626b56a5bf6e207d347f9badbd8e/py-polars/src/series.rs#L432-L434

I'll convert to draft.